### PR TITLE
View changelog

### PR DESCRIPTION
**Pull request by kntsoriano**
https://github.com/exercism/wip/issues/35

## Questions
1. Should we include the time the entry was published in here as well?
2. In the upcoming show page, should the URLs be slugs instead of ids? (e.g. `exercism.io/changelog/new-exercise-added`)

### CODE REVIEW

File: app/models/changelog_admin/null_referenceable.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+module ChangelogAdmin
+  class NullReferenceable
+    def icon
+      "https://https://via.placeholder.com/150"
```
Comment (kntsoriano) [inline app/models/changelog_admin/null_referenceable.rb]: This is the default icon when nothing is referenced in the changelog entry.

File: app/models/changelog_admin/referenceable_exercise.rb [hunk 2]
```diff
@@ -1,13 +1,19 @@
...
 
-    private
+    def title
+      "#{exercise.track_title} - #{exercise.title}"
+    end
 
-    def exercise
-      __getobj__
+    def icon
+      exercise.dark_icon_url
```
Comment (kntsoriano) [inline app/models/changelog_admin/referenceable_exercise.rb]: I'm unsure which icon to use.

File: app/models/changelog_admin/referenceable_track.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    def initialize(track)
+      @track = track
+    end
+
+    def title
+      track.title
+    end
+
+    def icon
+      track.bordered_green_icon_url
```
Comment (kntsoriano) [inline app/models/changelog_admin/referenceable_track.rb]: Same here. Unsure of which icon to use.

File: app/controllers/changelogs_controller.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+class ChangelogsController < ApplicationController
```
Comment (iHiD) [inline app/controllers/changelogs_controller.rb]: Could we call this `ChangelogEntriesController` pls?

File: app/models/changelog_admin/null_referenceable.rb (hunk 1 above)
Comment (iHiD) [inline app/models/changelog_admin/null_referenceable.rb]:
```suggestion
      "https://via.placeholder.com/150"
```

Comment (iHiD) [reply app/models/changelog_admin/null_referenceable.rb]: OK - cool. I'll work this out with Nic when we do styling.

File: config/routes.rb [hunk 5]
```diff
@@ -132,6 +132,8 @@
     end
   end
 
+  resource :changelog, only: [:show]
```
Comment (iHiD) [inline config/routes.rb]: I think this should be resources `:changelog_entries, only: [:index], path "changelog"`

Comment (kntsoriano) [reply config/routes.rb]: Sure. Will do.

**Comment by kntsoriano** [comment]
@iHiD Any thoughts on the questions above?

**Comment by iHiD** [comment]
@kntsoriano Missed those. Sorry.

1. Should we include the time the entry was published in here as well?

Yeah. That'd be nice. Using `time_ago_in_words` pls.

2. In the upcoming show page, should the URLs be slugs instead of ids? (e.g. exercism.io/changelog/new-exercise-added)

Good question. I like slugs, but I can see loads of slug clashes. Could you use slugs but posfixed with the id pls? So `exercism.io/changelog/new-exercise-added-#{id}`. That should be straight forward in FriendlyId I think.

**Pull request merged**